addons/purchase/models/purchase_order.py

```diff
@@ -579,6 +580,41 @@ def _add_supplier_to_product(self):
                 # supplier info should be added regardless of the user access rights
                 line.product_id.product_tmpl_id.sudo().write(vals)
 
+    def action_bill_matching(self):
+        return self.partner_id.open_purchase_matching()
+
+    def _prepare_down_payment_section_values(self):
+        context = {'lang': self.partner_id.lang}
+        res = {
+            'product_qty': 0.0,
+            'order_id': self.id,
+            'display_type': 'line_section',
+            'is_downpayment': True,
+            'sequence': (self.order_line[-1:].sequence or 9) + 1,
+            'name': _("Down Payments"),
+        }
+        del context
+        return res
+
+    def _create_downpayments(self, line_vals):
+        self.ensure_one()
+
+        # create section
+        if not any(line.display_type and line.is_downpayment for line in self.order_line):
+            section_line = self.order_line.create(self._prepare_down_payment_section_values())
+        else:
+            section_line = self.order_line.filtered(lambda line: line.display_type and line.is_downpayment)
+        vals = [
+            {
+                **line_val,
+                'sequence': section_line.sequence + i,
+            }
+            for i, line_val in enumerate(line_vals, start=1)
+        ]
+        downpayment_lines = self.env['purchase.order.line'].create(vals)
+        self.order_line += downpayment_lines
```

Comment:
```suggestion
        self.write({
            'order_line': [Command.create(val) for val in vals],
        })
```
Should even be integrated with the vals declaration above directly, I suppose?

Reply:
Same reason as above, we would need the downpayment lines afterwards

Reply:
same, I want to return only the created down payment lines